crates/core/c8y_api/src/smartrest/smartrest_serializer.rs

```diff
@@ -21,7 +22,12 @@ pub fn set_operation_executing(operation: impl C8yOperation) -> String {
 
 /// Generates a SmartREST message to set the provided operation to failed with the provided reason
 pub fn fail_operation(operation: impl C8yOperation, reason: &str) -> String {
-    fields_to_csv_string(&["502", operation.name(), reason])
+    if reason.len() <= 200 {
+        fields_to_csv_string(&["502", operation.name(), reason])
+    } else {
+        warn!("Failure reason too long, message trancuated to 200 characters");
```

Comment:
It might be good to log the whole message.

Reply:
Though we should be truncating the failure reason, and not ignoring it completely!

For example, ellipsis (`...`) can be used to represent that the failure reason was truncated:

```
My custom script output failed. check logs fo...
```

Reply:
Sorry ignore my comment, I didn't see the line below...

Reply:
> It might be good to log the whole message.

I would be careful logging the full message as this could be very large...in the end we also have the full stdout/stderr output in the log file under `/var/log/tedge/agent/*.log`

Reply:
Initial 100 characters were definitely too short so I increased that to 200 but I would go even further and set it to 500.